frontend/src/components/auth/TeacherSignup/TeacherSignup.tsx

```diff
@@ -46,47 +56,117 @@ const renderPageComponent = (
       return <></>;
   }
 };
+
 const TeacherSignup = (): React.ReactElement => {
-  const { authenticatedUser } = useContext(AuthContext);
-  const {
-    register,
-    handleSubmit,
-    setValue,
-    watch,
-    formState: { errors },
-  } = useForm<TeacherSignupForm>({ defaultValues, mode: "onChange" });
+  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+  const methods = useForm<TeacherSignupForm>({
+    defaultValues,
+    mode: "onChange",
+  });
   const [page, setPage] = React.useState(1);
+  const [user, setUser] = React.useState<AuthenticatedUser>();
+
+  const [registerTeacher] = useMutation<{ register: AuthenticatedUser }>(
+    REGISTER_TEACHER,
+  );
+  const [addTeacherToSchool] = useMutation<{
+    addTeacherToSchool: SchoolResponse;
+  }>(ADD_TEACHER_TO_SCHOOL, {
+    onCompleted() {
+      if (user) setAuthenticatedUser(user);
+    },
+  });
+  const [getSchool] = useLazyQuery<{ school: SchoolResponse }>(GET_SCHOOL, {
+    onCompleted: async (data) => {
+      await addTeacherToSchool({
+        variables: {
+          school: {
+            address: data.school.address,
+            city: data.school.city,
+            country: data.school.country,
+            name: data.school.name,
+            subRegion: data.school.subRegion,
+            teachers: data.school.teachers?.map(
+              (teacher: UserResponse) => teacher.id,
+            ),
+          },
+          schoolId: data.school.id,
+          teacherId: user?.id,
+        },
+      });
+    },
+  });
+  const [createSchool] = useMutation<{ createSchool: SchoolResponse }>(
+    CREATE_SCHOOL,
+    {
+      onCompleted() {
+        if (user) setAuthenticatedUser(user);
+      },
+    },
+  );
+
+  const onSubmitSuccess = async (data: TeacherSignupForm) => {
+    const newUser: AuthenticatedUser = await authAPIClient.register(
+      data.firstName,
+      data.lastName,
+      data.email,
+      data.password,
+      registerTeacher,
+    );
+    if (newUser) {
+      setUser(newUser);
+
+      if (data.school.id) {
```

Comment:
quick explanation on why i added the submission logic in this ticket

originally, the ticket was to rewrite `REGISTER_TEACHER` to handle all this logic , but i think due to its complexity, it's easier to handle them as separate calls.

i could create two mutations 
1. `REGISTER_TEACHER_NEW_SCHOOL` which combines `register` and `createSchool` 
2. REGISTER_TEACHER_ADD_TEACHER_TO_SCHOOL` which combines register`, `getSchool`, and `addTeacherToSchool` but I don't think there's a way to combine queries and mutations